### PR TITLE
Adds a new end-to-end test for Mistral 7b

### PR DESCRIPTION
**Pull request by shralex**
The changes include:
- A notebook in scratch_pad that runs the model from HF, for sample inputs and generates logits, which it stores in a file.
- A test that downloads the HF checkpoint, converts it to MaxText-compatible format, and uses it to do one forward pass on a sample input.
- Compares the logits obtained in both steps above to make sure they're equal. 

The old test, which asserted on the generated text being equal, rather than logits, is removed.

### CODE REVIEW

File: end_to_end/tpu/mistral/7b/test_mistral-7b.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# This file runs on daily basis and demonstrates:
```
Comment (khatwanimohit) [inline end_to_end/tpu/mistral/7b/test_mistral-7b.sh:3]: nit: mention where we run this test. v4-8 in this case.

**Review by khatwanimohit — APPROVED** [review]
LGTM!

File: end_to_end/tpu/mistral/7b/test_mistral-7b.sh
Comment (gobbleturk) [inline end_to_end/tpu/mistral/7b/test_mistral-7b.sh:1]: I guess this file should be split into 2 parts like the rest, but that is unrelated to fix here so okay to leave for now

**Review by gobbleturk — APPROVED** [review]

File: end_to_end/tpu/mistral/7b/test_mistral-7b.sh
Comment (shralex) [reply end_to_end/tpu/mistral/7b/test_mistral-7b.sh:1]: It was previously split, and merged into one by Mohit's request, since this is a small model.

Comment (gobbleturk) [reply end_to_end/tpu/mistral/7b/test_mistral-7b.sh:1]: ah makes sense, sounds good!

File: end_to_end/tpu/mistral/7b/test_mistral-7b.sh (hunk 1 above)
Comment (shralex) [reply end_to_end/tpu/mistral/7b/test_mistral-7b.sh]: Done
